.github/workflows/ci.yml

```
@@ -8,18 +8,37 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true    
 
+permissions:
+  id-token: write
+  contents: read
+
 jobs:
   main:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, idp]
 
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: 'Azuze CLI login'
```

Is it needed? (Get Secrets from Azure Key Vault will do it)

```suggestion
      - name: 'Azure CLI login'
```